### PR TITLE
Fix use before def errors during dispatch region formation.

### PR DESCRIPTION
**Pull request by MaheshRavishankar**

### CODE REVIEW

File: compiler/src/iree/compiler/Dialect/Flow/Transforms/RegionOpUtils.cpp [hunk 1]
```diff
@@ -212,7 +236,7 @@ FailureOr<Flow::DispatchRegionOp> Flow::appendDispatchRegionResult(
 }
 
 Flow::DispatchRegionOp Flow::makeDispatchRegionWithWorkload(
-    OpBuilder &builder, Location loc, Optional<ValueRange> workload) {
+    OpBuilder &builder, Location loc, Optional<SmallVector<Value>> workload) {
```
Comment (hanhanW) [inline compiler/src/iree/compiler/Dialect/Flow/Transforms/RegionOpUtils.cpp]: do we really need `workload` be Optional type? `SmallVector<Value>` is good enough to me.

File: compiler/src/iree/compiler/Dialect/Flow/Transforms/RegionOpUtils.cpp [hunk 2]
```diff
@@ -319,9 +347,19 @@ FailureOr<Flow::DispatchRegionOp> Flow::movePrecedingOpIntoDispatchRegion(
 FailureOr<Flow::DispatchRegionOp> Flow::wrapOpInDispatchRegion(
     RewriterBase &rewriter, Operation *op,
     Optional<Flow::WorkloadBuilder> workloadBuilder) {
-  Optional<ValueRange> workload = std::nullopt;
-  if (workloadBuilder.has_value()) workload = workloadBuilder->workload;
-  // Make an empty dispatch region right before the op.
+  Optional<SmallVector<Value>> workload = std::nullopt;
```
Comment (hanhanW) [inline compiler/src/iree/compiler/Dialect/Flow/Transforms/RegionOpUtils.cpp]: ditto, the check is happening at l.357 before using it. I think default can be empty list, and it is valid?

File: compiler/src/iree/compiler/Dialect/Flow/Transforms/RegionOpUtils.cpp (hunk 1 above)
Comment (MaheshRavishankar) [reply compiler/src/iree/compiler/Dialect/Flow/Transforms/RegionOpUtils.cpp]: Good point. Fixed.

**Review by hanhanW — APPROVED** [review]
LG, thanks!

**Pull request merged**